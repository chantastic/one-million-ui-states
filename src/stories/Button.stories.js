import { fn } from "@storybook/test";
import { Button } from "./Button";

/*
Lesson 1: Anatomy of a Story

Let's learn about stories by commenting out and re-writing sections of this file.
- Remove named exports (stories)
- Remove args (story args)
- Remove component args
- Remove component argsTypes
- Remove autodocs tag
- Remove parameters
- Remove title meta
- Remove component meta
- Write MVP story
*/

export default {
  // title: "Example/Button",
  component: Button,
  // parameters: {
  //   layout: "centered",
  // },
  // tags: ["autodocs"],
  // argTypes: {
  //   backgroundColor: { control: "color" },
  // },
  // args: { onClick: fn() },
};

export const Primary = {
  args: {
    primary: true,
    label: "Button",
  },
};

// export const Secondary = {
//   args: {
//     label: "Button",
//   },
// };

export const Large = {
  args: {
    size: "large",
    label: "Button",
  },
};

// export const Small = {
//   args: {
//     size: "small",
//     label: "Button",
//   },
// };

/*
Lesson 2: Compose stories

Stories (using SCF object syntax) can compose args from other stories.
Uncomment (or re-write) the story below to make a new story from others.
*/

export const LargePrimary = {
  args: {
    ...Large.args,
    ...Primary.args,
  },
};
