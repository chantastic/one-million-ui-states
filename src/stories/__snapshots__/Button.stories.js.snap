// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories/Button Large smoke-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "name": "Button",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "stories-button--large",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/Button LargePrimary smoke-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "name": "Button",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "stories-button--large-primary",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/Button Primary smoke-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "name": "Button",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "stories-button--primary",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
